src/components/Autocomplete.js

```diff
@@ -1,63 +1,51 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import * as SmartySDK from "smartystreets-javascript-sdk";
 import * as sdkUtils from "smartystreets-javascript-sdk-utils";
 import InputForm from "./InputForm";
 import Suggestions from "./Suggestions";
 
-export default class Autocomplete extends React.Component {
-	constructor(props) {
-		super(props);
-
-		this.state = {
-			lookup: new SmartySDK.usAutocompletePro.Lookup,
-			client: this.autoCompleteClient,
-			shouldValidate: true,
-			freeform: "",
-			address1: "",
-			address2: "",
-			city: "",
-			state: "",
-			zipCode: "",
-			country: "US",
-			suggestions: {result: []},
-			error: "",
-		};
-
-		const SmartyCore = SmartySDK.core;
-		const websiteKey = ""; // Your website key here
-		const smartySharedCredentials = new SmartyCore.SharedCredentials(websiteKey);
-
-		this.SmartyCore = SmartyCore;
-		this.autoCompleteClient = new SmartyCore.ClientBuilder(smartySharedCredentials).withLicenses(["us-autocomplete-pro-cloud"]).buildUsAutocompleteProClient();
-		this.internationalautocompleteClient = new SmartyCore.ClientBuilder(smartySharedCredentials).withLicenses(["international-autocomplete-v2-cloud"]).buildInternationalAddressAutocompleteClient();
-		this.usStreetClient = new SmartyCore.ClientBuilder(smartySharedCredentials).buildUsStreetApiClient();
-		this.internationalStreetClient = new SmartyCore.ClientBuilder(smartySharedCredentials).buildInternationalStreetClient();
-
-		this.updateField = this.updateField.bind(this);
-		this.updateCheckbox = this.updateCheckbox.bind(this);
-		this.queryAutocompleteForSuggestions = this.queryAutocompleteForSuggestions.bind(this);
-		this.selectSuggestion = this.selectSuggestion.bind(this);
-		this.updateStateFromValidatedUsAddress = this.updateStateFromValidatedUsAddress.bind(this);
-		this.validateAddress = this.validateAddress.bind(this);
-		this.formatAutocompleteSuggestion = this.formatAutocompleteSuggestion.bind(this);
-	}
-
-	updateStateFromForm(key, value) {
-		const newState = {};
-		newState[key] = value;
-
-		this.setState(newState);
-	}
-
-	updateField(e) {
-		this.updateStateFromForm(e.target.id, e.target.value);
-	}
-
-	updateCheckbox(e) {
-		this.updateStateFromForm(e.target.id, e.target.checked);
-	}
-
-	formatAutocompleteSuggestion(suggestion) {
+const Autocomplete = () => {
+	const [state, setState] = useState({
```

Comment:
I think it would be best if instead of having one big state for everything, to split them up. Or group the form fields together into one state, but then separate everything else. What are your thoughts?

Reply:
I did some refactoring here, which ended up taking it's own shape based on how to app is laid out.

For the most part, I tried to keep the formValues together and made new state variables for the other things.